src/idl_gen_cpp.cpp

```diff
@@ -307,6 +307,24 @@ class CppGenerator : public BaseGenerator {
       }
     }
 
+    // Check if enum arrays are used without specifying --scoped-enums
+    if (!parser_.opts.scoped_enums) {
+      for (auto it_s = parser_.structs_.vec.begin();
+           it_s != parser_.structs_.vec.end(); ++it_s) {
+        const auto &struct_def = **it_s;
+        for (auto it_f = struct_def.fields.vec.begin();
+             it_f != struct_def.fields.vec.end(); ++it_f) {
+          const auto &field_def = **it_f;
+          const auto &field_type = field_def.value.type;
+
+          if (IsArray(field_type) && IsEnum(field_type.VectorType())) {
+            printf("--scoped-enums must be enabled to use enum arrays\n");
+            return false;
+          }
+        }
+      }
+    }
+
```

Comment:
Why do you need to traverse `parser_.structs_.vec`?
Is it possible to use lazy checking? For example after line 2745 when a field of a struct is generated.